modules/core/src/main/scala/gem/Sequence.scala

```diff
@@ -3,29 +3,45 @@ package gem
 import scalaz._
 import Scalaz._
 
+case class Sequence[S](
+  id: Sequence.Id,
+  steps: List[S])
+
 object Sequence {
 
   trait Id {
     def oid: Observation.Id
     def name: String
   }
 
-  object Id extends ((Observation.Id, String) => Option[Sequence.Id]) {
-    def apply(o: Observation.Id, n: String): Option[Sequence.Id] =
+  object Id {
+    def fromName(o: Observation.Id, n: String): Option[Sequence.Id] =
```

Comment:
`Id` isn't a `case class` because not just any `name` `String` is valid.  I think there's some trick with `sealed abstract case class` that lets you control access to the constructor?  So there might be a better way to do this.